activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -632,12 +632,30 @@ def type_cast_calculated_value(value, operation, type)
       def select_for_count
         if select_values.present?
           return select_values.first if select_values.one?
-          select_values.join(", ")
+
+          select_values.map do |field|
+            column = arel_column(field.to_s) do |attr_name|
+              Arel.sql(attr_name)
+            end
+
+            if column.is_a?(Arel::Nodes::SqlLiteral)
+              column
+            else
+              "#{adapter_class.quote_table_name(column.relation.name)}.#{adapter_class.quote_column_name(column.name)}"
+            end
+          end.join(", ")
```

Comment:
As with the models used in the added test case (which have duplicate column names), we need to be explicit to reference the table name too to not get the ambiguous column error.